### DISCUSSION TITLE
about I use Parallel Routes report 404 not found

### DISCUSSION THREAD

**Discussion by duoluodexiaoxiaoyuan** [vercel/next.js#55090] [App Router]
I recently study next.js, I very like it. I meet a problem.

Detail Message:
win11
node v16.18.0
next13.4.2

Main code
app -> layout.js
```javascript
import './globals.css'
import { Inter } from 'next/font/google'
import { Suspense } from 'react'
import Loading from '@/app/loading.js'

const inter = Inter({ subsets: ['latin'] })

export const metadata = {
  title: 'Create Next App',
  description: 'Generated by create next app',
}
// 默认写的是{children}，但是我们写成props的时候要把{}去掉
export default function RootLayout(props) {
  return (
    <html lang="en">
      <body className={inter.className}>
        {props.children}
        {props.team}
        {props.analytics}
        {/* <Suspense fallback={<Loading/>}>
          {props.children}
          {props.team}
          {props.analytics}
        </Suspense> */}
      </body>
    </html>
  )
}

```

Directory Structure

![image](https://github.com/vercel/next.js/assets/59086082/22a4255d-c8fa-4b77-b55c-80f4c3dfc696)

this is my repo： https://github.com/duoluodexiaoxiaoyuan/mynextproject

**Comment by devjiwonchoi** [comment]
How can I reproduce 404? It seems like it's working fine.

**Comment by duoluodexiaoxiaoyuan** [comment]
> How can I reproduce 404? It seems like it's working fine.

yeah, I fell is true.But it's actually fail. this is my repo: https://github.com/duoluodexiaoxiaoyuan/mynextproject. Can u check up, thank u.

**Reply by devjiwonchoi** [reply]
@duoluodexiaoxiaoyuan yes I did run dev your source and is not showing 404. Can you share your condition on how to reproduce it?

![Screenshot 2023-09-07 at 4 42 16 PM](https://github.com/vercel/next.js/assets/120007119/e1cb3f38-0835-4313-ae80-c9b095c152b2)
![Screenshot 2023-09-07 at 4 42 06 PM](https://github.com/vercel/next.js/assets/120007119/f7edd9d3-b36d-4bca-9cbb-e4829817c013)

**Comment by duoluodexiaoxiaoyuan** [comment]
> @duoluodexiaoxiaoyuan yes I did run dev your source and is not showing 404. Can you share your condition on how to reproduce it?
> 
> ![Screenshot 2023-09-07 at 4 42 16 PM](https://user-images.githubusercontent.com/120007119/266242565-e1cb3f38-0835-4313-ae80-c9b095c152b2.png) ![Screenshot 2023-09-07 at 4 42 06 PM](https://user-images.githubusercontent.com/120007119/266242576-f7edd9d3-b36d-4bca-9cbb-e4829817c013.png)

thank u, When I tried again, it's ok. it's amazing. I solve this problem  four hours.I don't know why ?

**Reply by devjiwonchoi** [reply]
Glad it's working:) Lmk if it breaks again.

**Comment by duoluodexiaoxiaoyuan** [comment]
> 
I update my repo. Now I have extreal problem.

![image](https://github.com/vercel/next.js/assets/59086082/f373430f-885b-4561-8ab6-3b8059d0cfd7)

**Reply by devjiwonchoi** [reply]
![Screenshot 2023-09-07 at 4 57 35 PM](https://github.com/vercel/next.js/assets/120007119/eacd9dc5-5c56-4f68-b217-e899bf5c0c1d)

The folder structure seems weird. Is it intended?

**Reply by duoluodexiaoxiaoyuan** [reply]
![image](https://github.com/vercel/next.js/assets/59086082/23a4ea5d-6c18-4d50-b335-0814b345a339)

this picture come from video

**Comment by duoluodexiaoxiaoyuan** [comment]
> 
yes, I see a videos about next.js.

**Reply by devjiwonchoi** [reply]
Which video is it can you share?

**Comment by duoluodexiaoxiaoyuan** [comment]
> Which video is it can you share?
thank u again, I'm very happy to meet  you here.
https://www.bilibili.com/video/BV1do4y1K7BF?p=11&vd_source=90a9c7618953f5d275200b2fa7aed461

**Reply by devjiwonchoi** [reply]
> thank u again, I'm very happy to meet you here.

No problem, I'm here to help:)

Ok, so it receives dynamic params to find the image, so when you're on `/`, the params is [] which will not bring any photo, resulting in 404.

However, http://localhost:3000/photos/1 displays photo id 1.

**Reply by duoluodexiaoxiaoyuan** [reply]
final question, when i use props.modal, project start error, I don't how to slove

**Reply by duoluodexiaoxiaoyuan** [reply]
previous problem is slove, i curcious when i add  {props.modal} , page display 404

**Reply by devjiwonchoi** [reply]
TBH I don't understand what `(.)photos/[id]` can do. If you could elaborate for me it'd be great. Therefore I looked into the video and it says `// TODO: 判断`, I'm assuming it's gonna be covered on the next lecture?

**Comment by duoluodexiaoxiaoyuan** [comment]
> 

thank u, I found problem. I should in app/page.js     photos.map photo

**Comment by devjiwonchoi** [comment]
I'd recommend you close this as answered and open a new discussion for the next question that comes up. Also, I'd recommend you use CodeSandBox so as to let us check it out faster and more conveniently.

**Comment by duoluodexiaoxiaoyuan** [comment]
> I'd recommend you close this as answered and open a new discussion for the next question that comes up. Also, I'd recommend you use CodeSandBox so as to let us check it out faster and more conveniently.我建议您在回答后关闭此对话框，并为出现的下一个问题打开新的讨论。此外，我建议您使用CodeSandBox，以便让我们更快，更方便地检查它。
yeah, very thank. a good boy is help slove. thank you very much
https://github.com/duoluodexiaoxiaoyuan/mynextproject/pull/2
